src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -420,51 +456,87 @@ object DeployPlugin extends AutoPlugin {
-    * containing all the relevant deployment keys.
-    *
-    * Exits the program if any of the keys required for deployment are missing.
-    *
+  /** Container for config needed to deploy a project to a host.
+    * @param host The fully-qualified name of the host the project should be deployed to
+    * @param directory The directory on the host in which the project's code should be placed
+    * @param sshUser The username to use when accessing the host
+    * @param startupScript The path, relative to the deploy directory, to the script that should
+    *                      be used to restart the project after deploying the new code
+    * @param startupArgs Arguments to pass to the startup script
+    */
+  case class HostConfig(
+    host: String,
+    directory: String,
+    sshUser: String,
+    startupScript: String,
+    startupArgs: Seq[String]
+  )
+
+  /** Wrapper for many [[org.allenai.plugins.DeployPlugin.HostConfig]] objects, along with any data
+    * that should be shared between them, when deploying a project.
+    * @param hostConfigs configuration objects specifying how the project should be deployed to
+    *                    many hosts
+    * @param projectVersion the version of the project being deployed
+    */
+  case class DeployConfig(hostConfigs: Seq[HostConfig], projectVersion: Option[String])
+
+  /** Transform the given [[com.typesafe.config.Config]] object into a corresponding
+    * [[org.allenai.plugins.DeployPlugin.DeployConfig]] object.
     * @param targetName the name of the target to print on error
-    * @param targetConfig the config to parse
+    * @param targetConfig env-level deployment config to convert
     */
-  def validateAsMap(targetName: String, targetConfig: Config): Map[String, String] = {
-    // Validate that the target has all the required environment variables.
-    val requiredEnv = Seq("AWS_PEM_FILE")
-    for (variable <- requiredEnv) {
-      val value = sys.env.get(variable).getOrElse {
-        throw new IllegalArgumentException(s"Environment variable ${variable} undefined.")
+  def parseConfig(targetName: String, targetConfig: Config): DeployConfig = {
+
+    DeployConfig(
+      hostConfigs = Seq(parseHostConfig(targetConfig.getConfig("deploy"))),
+      projectVersion = try {
+        Some(targetConfig.getString("project.version"))
+      } catch {
+        case _: ConfigException.Missing => None
+        case _: ConfigException.WrongType =>
+          throw new IllegalArgumentException(s"Error: 'project.version' must be a string")
       }
+    )
+  }
 
-      require(!value.isEmpty, "Environment variable ${value} is empty.")
-    }
+  /** Transform the given [[com.typesafe.config.Config]] object into a corresponding
+    * [[org.allenai.plugins.DeployPlugin.HostConfig]] object.
+    * @param hostConfig host-level config to convert
+    */
+  def parseHostConfig(hostConfig: Config): HostConfig = {
+    val renderedConfig = hostConfig.root().render()
```

Comment:
You can also inline this below where it's needed. It's only in two places - and it's only two function calls.